### ISSUE TITLE
Some notes while browsing the code

### ISSUE THREAD

**Issue by leplatrem** [Kinto/kinto-admin#204]
Some opinionated remarks:

* We could just use `id` instead of `name` in forms. This cosmestic detail make the code a bit complex to follow sometimes (https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/reducers/bucket.js#L38 https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/components/BucketCreateForm.js#L20)
* We should use `bid` instead of `bucket` if it is contains just an id (cf. https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/sagas/route.js#L77 https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/reducers/collection.js#L56)
* We should not «merge» state properties with object data, this makes code (and object destructing) sometimes hard to follow. Especially if this value can be obtained via `state.params` ;) (c.f. https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/types.js#L59 https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/sagas/route.js#L77 https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/reducers/collection.js#L56)
* The type `Collection` should be name `CollectionState`, otherwise it's confusing with `CollectionData`
* ~~`CollectionList` should be renamed `CollectionRecords`~~ (#219)
* ~~Since we have the URLs `create-bucket` and `create-collection` (and `create-group` in #202), we should have `create-record` instead of just `add`~~ (#219)
* Since we have `/buckets/:bid/collections/:cid/edit` we should have `/buckets/:bid/collections/:cid/records/:rid/edit`
* Since we have `/buckets/:bid/collections` (cf #199), we should have `/buckets/:bid/collections/:cid/records` instead of `/buckets/:bid/collections/:cid`
* ~~`BucketCreateForm.js` and `BucketEditForm.js` should be renamed `BucketCreat.js` and `BucketEdit.js`~~ (#219)
*  For JSON fields, it would be nice if the stringification and parsing would be done in the same place (both at the form level) : https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/components/BucketForm.js#L79-L80  https://github.com/Kinto/kinto-admin/blob/v1.3.0/scripts/components/BucketEditForm.js#L31-L32

![](http://i.giphy.com/8p8E1sylIARDW.gif)

**Comment by n1k0** [comment]
Updated the description as #219 addressed a few listed items.

**Comment by n1k0** [comment]
#231 will take care of the rest.

**Comment by n1k0** [comment]
These have been fixed by #231 except the JSONEditor refactoring for wich I've filed #204.
